core/client/app/components/gh-selectize.js

```diff
@@ -106,6 +82,33 @@ export default EmberSelectizeComponent.extend({
         });
 
         this.set('selection', reorderedSelection);
+    },
+
+    _preventOpeningWhenBlank() {
+        let openOnFocus = this.get('openOnFocus');
+
+        if (!openOnFocus) {
+            run.schedule('afterRender', this, function () {
+                let selectize = this._selectize;
+                if (selectize) {
+                    selectize.on('dropdown_open', function () {
+                        if (isBlank(selectize.$control_input.val())) {
+                            selectize.close();
+                        }
+                    });
+                    selectize.on('type', function (filter) {
+                        if (isBlank(filter)) {
+                            selectize.close();
+                        }
+                    });
+                }
+            });
+        }
+    },
+
+    didInsertElement() {
+        this._super(...arguments);
+        this._preventOpeningWhenBlank();
```

Comment:
I did it this way as to me it reads better and gives an at-a-glance indication of what's being done during the  initialisation step which was the main reason for having the `_dontOpenWhenBlank: on('didInsertElement' ...` syntax in the first place.